### ISSUE TITLE
Add integration/mocked tests with libnss_wrapper

### ISSUE THREAD

**Issue by apognu** [rustadopt/uzers-rs#14]
As outlined in #13 and in the code, unit tests cannot actually be complete because because we (rightly) cannot assume a user is present (and configured the way tests expect) on the system running the tests, and creating real users to run the tests is hard to justify.

There might be a solution by using `nss-wrapper`, a library that should be loaded with `LD_PRELOAD` that could be used to provide fixtures for various NSS entities (including users and groups).

It is still not completely ideal because it requires installing this library and running the tests with the right environment variables, but it could at the minimum be a good thing for CI (a prototype can be seen on [this branch](https://github.com/apognu/uzers-rs/tree/dev/nss-wrapper-tests) and in [this action step](https://github.com/apognu/uzers-rs/actions/runs/6650088259/job/18069624388#step:6:95).

Basically:

 * Added (dummy, for the prototype) integration tests, gated behind a feature, so they do not run by default on `cargo test`, otherwise they would fail without the fixtures
 * Added two fixtures in the `tests/fixtures/` directory for `passwd` and `group`
 * In the action, install `libnss-wrapper` and set environment variables that point to the fixtures
 * Add a test step, running the integration tests (with the gate feature) and `LD_PRELOAD`.

I probably will not be able to retroactively write all the necessary tests for this library, but this could at least provide the necessary framework to do so.

What do you think?

**Comment by gierens** [comment]
Ah interesting, that's definitely a good hint ... in eza we use nix for the sandboxing but I'm not a nix person so much at lest not yet so I was hesitant to introduce this here. This seems like a lightweight alternative for this kind of thing. Thanks!

**Comment by apognu** [comment]
I will clean my prototype, write a few example tests (both unit and integration) to pave the way, but functionally, it already works in CI on my branch. So I'll create a PR.

**Comment by apognu** [comment]
I'll close this, it was merged very recently. 🎉

**Comment by gierens** [comment]
Ah yes thanks! We forgot to put a closing keyword in the PR :innocent:
